e2e/stack.go

```diff
@@ -53,89 +51,139 @@ func New(
 	monomerEngineURL,
 	monomerCometURL,
 	opNodeURL *url.URL,
-	contractsRootDir string,
-	l1BlockTime time.Duration,
+	deployConfigDir string,
+	l1stateDumpDir string,
+	l1BlockTime uint64,
 	eventListener EventListener,
 ) *Stack {
 	return &Stack{
-		anvilURL:         anvilURL,
 		monomerEngineURL: monomerEngineURL,
 		monomerCometURL:  monomerCometURL,
 		opNodeURL:        opNodeURL,
-		contractsRootDir: contractsRootDir,
+		deployConfigDir:  deployConfigDir,
+		l1stateDumpDir:   l1stateDumpDir,
 		eventListener:    eventListener,
 		l1BlockTime:      l1BlockTime,
 	}
 }
 
+// helper shim to allow decoding of hex nonces
+type auxDumpAccount struct {
+	Balance     string                 `json:"balance"`
+	Nonce       string                 `json:"nonce"`
+	Root        hexutil.Bytes          `json:"root"`
+	CodeHash    hexutil.Bytes          `json:"codeHash"`
+	Code        hexutil.Bytes          `json:"code,omitempty"`
+	Storage     map[common.Hash]string `json:"storage,omitempty"`
+	Address     *common.Address        `json:"address,omitempty"` // Address only present in iterative (line-by-line) mode
+	AddressHash hexutil.Bytes          `json:"key,omitempty"`     // If we don't have address, we can output the key
+}
+
+// helper shim to allow decoding of hex nonces
+type auxDump struct {
+	Root     string                    `json:"root"`
+	Accounts map[string]auxDumpAccount `json:"accounts"`
+	// Next can be set to represent that this dump is only partial, and Next
+	// is where an iterator should be positioned in order to continue the dump.
+	Next []byte `json:"next,omitempty"` // nil if no more accounts
+}
+
+func (d *auxDump) ToStateDump() (*state.Dump, error) {
+	accounts := make(map[string]state.DumpAccount)
+
+	for k, v := range d.Accounts { //nolint:gocritic
+		nonce, err := hexutil.DecodeUint64(v.Nonce)
+		if err != nil {
+			return nil, fmt.Errorf("decode nonce: %v", err)
+		}
+
+		accounts[k] = state.DumpAccount{
+			Balance:  v.Balance,
+			Nonce:    nonce,
+			Root:     v.Root,
+			CodeHash: v.CodeHash,
+			Code:     v.Code,
+			Storage:  v.Storage,
+		}
+	}
+
+	return &state.Dump{
+		Root:     d.Root,
+		Accounts: accounts,
+	}, nil
```

Comment:
Should the `Next` field be populated here as well if it's not nil or does that not matter for the e2e test?

Reply:
Next is indeed `nil` / empty, but not populating it seems like a potential future bug that should be avoided. 6526c4d